src/TransCreator/PageFooter.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import moment from 'moment';
+import {
+  Text,
+  Button,
+  Footer,
+  FooterTab,
+  Toast,
+} from 'native-base';
+import { Platform } from 'react-native';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import styles from './Style';
+import { addNewTransaction } from '../Stats/actionCreator';
+
+const pressHandler = (amount, date, view, setTransAmount, setTransType, setTransDate, dispatch) => {
+  if (!amount) {
+    Toast.show({
+      position: 'top',
+      textStyle: { color: '#F4511E' },
+      text: 'Please enter the amount',
+      buttonText: 'Okay',
+    });
+    return;
+  }
+
+  setTransAmount(null);
+  setTransType('Expense');
+  setTransDate(moment().unix());
+  dispatch(addNewTransaction(view, date, amount));
+};
+
+const PageFooter = ({
+  transAmount, transType, transDate, setTransAmount, setTransType, setTransDate,
```

Comment:
PageFooter shouldn't care about these values and methods, it should just simple get an onSubmit method from the parent and call it. The parent already knows these values and the footer cannot change them

Reply:
fixed by passing handler function